src/toil/jobStores/awsJobStore.py

```diff
@@ -829,35 +653,25 @@ def fromItem( cls, item, jobStoreID=None ):
         :type item: Item
         :rtype: AWSJob
         """
-        if jobStoreID is None: jobStoreID = item.name
-        try:
-            del item[ 'parentJobStoreID' ]
-        except KeyError:
-            pass
-        item = { k: cls.fromItemTransform[ k ]( v ) for k, v in item.iteritems( ) }
-        return cls( jobStoreID=jobStoreID, **item )
-
-    toItemTransform = defaultdict( lambda: passThrough,
-                                   command=fromNoneable,
-                                   jobStoreID=skip,
-                                   updateID=str,
-                                   children=skip,
-                                   stack=lambda v: fromList( map( repr, v ) ),
-                                   logJobStoreFileID=fromNoneable,
-                                   predecessorsFinished=fromSet,
-                                   jobsToDelete=fromList ,
-                                   predecessorNumber=str,
-                                   remainingRetryCount=str)
+        chunkedJob = item.items()
+        chunkedJob.sort()
+        if len(chunkedJob)==1:
+            wholeJobString = chunkedJob[0][1] #first element of list = tuple, second element of tuple = serialized job
+        else:
+            wholeJobString = ''.join(item[1] for item in chunkedJob)
+        return cPickle.loads(bz2.decompress(base64.b64decode(wholeJobString)))
 
     def toItem( self, parentJobStoreID=None ):
         """
         :rtype: Item
         """
-        item = self.toDict( )
-        if parentJobStoreID is not None:
-            item[ 'parentJobStoreID' ] = parentJobStoreID
-        item = ((k, self.toItemTransform[ k ]( v )) for k, v in item.iteritems( ))
-        return { k: v for k, v in item if v is not None }
+        item = {}
+        serializedAndEncodedJob = base64.b64encode(bz2.compress(cPickle.dumps(self)))
+        # this convoluted expression splits the string into chunks of 1024 - the max value for an attribute in SDB
+        jobChunks = [serializedAndEncodedJob[i:i+1024] for i in range(0, len(serializedAndEncodedJob), 1024)]
```

Comment:
And this is the right way to partition a string. The same combination of slicing and range() with non-unit increments can be used to partition a list, too.